Better Explain State Across All Develop (#1251)
near/docs: docs/2.develop/deploy.md

Let us guide you on how to use the [NEAR CLI](../4.tools/cli.md) to deploy your contract
and call its methods.

:::info

> In this page we will only touch on the basics on NEAR CLI. For more information visit the

Reviewer: ```suggestion
On this page, we will only cover the basics of NEAR CLI. For more information visit the
```

[NEAR CLI documentation page](../4.tools/cli.md).
:::

---

## Deploying the Contract

Thanks to the `NEAR CLI` deploying a contract is as simple as:

1. Compiling the contract to wasm (done automatically through `yarn build` in our templates).
2. Deploy it into the desired account using the [NEAR CLI](../4.tools/cli.md#near-deploy):

#### Create an Account and Deploy
```bash
# Automatically deploy the wasm in a new account
near dev-deploy <route_to_wasm>

# Get the account name
cat ./neardev/dev-account
```

#### Deploy in an Existing Account
```bash
# login into your account
near login

# deploy the contract
near deploy <accountId> <route_to_wasm>
```

:::tip
You can overwrite a contract by deploying another on top of it. In this case, the account's logic
will change, but the state will persist
:::

:::info

> By default `near-cli` uses the `testnet`. Define `NEAR_ENV=mainnet` to deploy into `mainnet`.

Reviewer: ```suggestion
By default `near-cli` uses the `testnet` network. Define `NEAR_ENV=mainnet` to deploy into `mainnet`.
```
